### ISSUE TITLE
QoL Improvements

### ISSUE THREAD

**Issue by fmbearmf** [TheRealHiThere/Basic-Python-Chess-Engine#3]
# Just a few things that **should** be added, but are low priority.

- [x] Main Menu, preferably with settings
- [x] A sound engine
- [x] AI opponent
- [ ] Multiplayer

## Main Menu

Just a play, settings, and quit button should be fine.

## Sound Engine

As the name implies, support for music.

## AI Opponent

An algorithm or AI for single-player.

## Multiplayer

Possibly P2P networking, though this seems a little unlikely:
[Paper on why networking would be challenging, but also how it could be done](https://pdos.csail.mit.edu/papers/p2pnat.pdf#:~:text=Network%20Address%20Translation%20%28NAT%29%20causes%20well-known%20dif%EF%AC%81culties%20for,about%20their%20robustness%20or%20relative%20merits%20is%20slimmer.)

TL;DR
Network Address Translation (NAT) causes well-known
difficulties for peer-to-peer (P2P) communication, since
the peers involved may not be reachable at any globally
valid IP address.
